### ISSUE TITLE
Bug: [space-infix-ops] and [type-annotation-spacing] are in conflict

### ISSUE THREAD

**Issue by hasban12138** [typescript-eslint/typescript-eslint#5115]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

null

### Repro Code

```typescript
private word: {[key: string]: any} = null;
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "space-infix-ops": "off",
    "@typescript-eslint/space-infix-ops": ["error"],
    "@typescript-eslint/type-annotation-spacing": ["error"]
  },
};
```

### Expected Result

it may like this:
` private word: {[key: string]: any} = null;`

### Actual Result

`space-infix-ops` and `type-annotation-spacing` are in conflict
it maybe this one https://github.com/typescript-eslint/typescript-eslint/issues/5041 in 5.27.0

![QQ截图20220531112505](https://user-images.githubusercontent.com/33484052/171087871-56ad3ad9-dd23-43e4-88b5-2cdb5b20be3b.png)

![QQ截图20220531112520](https://user-images.githubusercontent.com/33484052/171087875-7e918a99-89d0-4b48-b583-08e080f5bce7.png)

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.27.0` |
| `@typescript-eslint/parser`            | `5.27.0` |
| `TypeScript`                           | `4.5.5` |
| `ESLint`                               | `8.9.0` |
| `node`                                 | `14.17.6` |

**Comment by armano2** [comment]
fixed by #5113, can you check canary 5.27.1-alpha.5

**Comment by hasban12138** [comment]
@armano2  thanks
